### PR TITLE
Falling down a z-level while standing up can (occasionally) break your legs. Felinids will now always land on their feet (for better or for worse). 

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

- Falling down a z-level while standing up will apply the damage directly to your legs. 
   - This damage has the chance of rolling wounds, so falling z-levels -> broken legs. 
   - If you are lying down when you fall, the damage is spread through your body like normal. 

- Felinids who fall down a z-level while standing will always land on their feet. 
   - Instead of being floored for 5 seconds / levels fallen, you will receive a slowdown for 2 seconds / levels fallen. 
   - However, while they have cat reflexes, they do not have cat legs. Landing on your feet will deal additional damage. 
      - For falling 1 z-level, this is ~18 damage (instead of ~12).
      - For 2 z-levels, this is ~50 damage (instead of ~32). 

- Cats (basic mobs) are capable of falling down z-levels without sustaining damage. 

image

## Why It's Good For The Game

Add some more depth to falling down. (No pun intended)

Rather than dropping down a level and walking it off after getting up, you might dislocate your leg. 

Also adds some more species interactions. Felines might be able to give people the slip easier when dropping down into the undertram, thanks to their cat-ness. 

## Changelog

:cl: Melbert
add: Falling down a z-level while standing now damages your legs rather than your entire body. This also means falling down multiple z-s may rarely break your legs. 
add: Felinids now land on their feet if they fall down a z-level while standing. This replaces the knockdown with a short slowdown, and also has the trade-off of causing more leg damage. 
add: Cats can now fall z-levels without sustaining damage. 
/:cl:

### CODE REVIEW

**Comment by MrMelbert** [comment]
> Do moth wings still prevent fall damage entirely?

moth wings only prevent fall damage if falling one level or less, and that aspect is untouched

File: code/modules/mob/living/basic/pets/cat/cat.dm [hunk 1]
```diff
@@ -56,6 +56,7 @@
 	AddElement(/datum/element/pet_bonus, "purrs!")
 	add_verb(src, /mob/living/proc/toggle_resting)
 	ADD_TRAIT(src, TRAIT_VENTCRAWLER_ALWAYS, INNATE_TRAIT)
+	ADD_TRAIT(src, TRAIT_CATLIKE_GRACE, INNATE_TRAIT)
```
Comment (san7890) [inline code/modules/mob/living/basic/pets/cat/cat.dm:59]:
```suggestion
	add_traits(list(TRAIT_CATLIKE_GRACE, TRAIT_VENTCRAWLER_ALWAYS), INNATE_TRAIT)
```

**Review by san7890 — CHANGES_REQUESTED** [review]

**Comment by excalisword** [comment]
Does space ninja have a chance to receive wounds when they fall?

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -46,19 +46,74 @@
...
+			visible_message(
+				span_notice("[src] makes a hard landing on [impacted_turf], but lands safely on [p_their()] feet!"),
+				span_notice("You make a hard landing on [impacted_turf], but land safely on your feet!"),
+			)
+			new /obj/effect/temp_visual/mook_dust/small(impacted_turf)
+			return .
+
+		incoming_damage *= 1.66
+		add_movespeed_modifier(/datum/movespeed_modifier/landed_on_feet)
+		addtimer(CALLBACK(src, TYPE_PROC_REF(/mob, remove_movespeed_modifier), /datum/movespeed_modifier/landed_on_feet), levels * 2 SECONDS)
```
Comment (necromanceranne) [inline code/modules/mob/living/living.dm:95]: why not the staggered status effect? Saves having to add a callback.

**Comment by MrMelbert** [comment]
> Does space ninja have a chance to receive wounds when they fall?

ninja is... entirely untouched? not even sure why you'd bring it up

File: code/modules/mob/living/living.dm (hunk 2 above)
Comment (MrMelbert) [reply code/modules/mob/living/living.dm:95]: staggered seems to have a set slowdown value and it's much faster than the slowdown value being used here

File: code/_globalvars/traits/_traits.dm [hunk 3]
```diff
@@ -212,6 +213,7 @@ GLOBAL_LIST_INIT(traits_by_type, list(
 		"TRAIT_GREENTEXT_CURSED" = TRAIT_GREENTEXT_CURSED,
 		"TRAIT_GUN_NATURAL" = TRAIT_GUN_NATURAL,
 		"TRAIT_GUNFLIP" = TRAIT_GUNFLIP,
+		"TRAIT_CATLIKE_GRACE" = TRAIT_CATLIKE_GRACE,
```
Comment (san7890) [inline code/_globalvars/traits/_traits.dm:216]: why'd you add it twice to the same list

**Review by san7890 — CHANGES_REQUESTED** [review]

**Comment by MrMelbert** [comment]
> does this effect the freerunning quirk?

same as wings, untouched

**Comment by ATH1909** [comment]
Not entirely a fan of felinids taking MORE fall damage than humans; humans don't have cat legs either, so why are felinids taking more damage for not having them? I was kind of expecting the "always land on their feet" effect to be that they always take fall damage to their legs, regardless of whether or not they were standing.

**Comment by san7890** [comment]
if we simulated ribs breaking and causing internal bleeding this would be way more favorable to felinids for the record

**Comment by moocowswag** [comment]
can things still cushion the fall and negate this?

**Pull request merged**